hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -123,4 +125,11 @@ private Object getFieldValueFromInternalRow(InternalRow row, Schema recordSchema
       return null;
     }
   }
+
+  @Override
+  public UnaryOperator<InternalRow> projectRecord(Schema from, Schema to) {
+    UnsafeProjection projection = HoodieInternalRowUtils.getCachedUnsafeProjection(AvroConversionUtils.convertAvroSchemaToStructType(from),
```

Comment:
Use `HoodieInternalRowUtils.getCachedSchema(schema)` to avoid parsing Avro schema per record?

Reply:
HoodieCatalystExpressionUtils.generateUnsafeProjection is what I wanted to use, but the file contains both an object and a trait with the name HoodieCatalystExpressionUtils. When using scala2.11 and calling with java, it fails building with a no such method exception.

Reply:
I was referring to `AvroConversionUtils.convertAvroSchemaToStructType` which can be replaced by `HoodieInternalRowUtils.getCachedSchema(schema)` to avoid repeated parsing.